bindings/nodejs/__test__/index.spec.mjs

```diff
@@ -15,20 +15,72 @@
 
 import test from 'ava'
 
-import { OperatorFactory } from '../index.js'
+import { Memory } from '../index.js'
 
 test('test memory write & read', async (t) => {
-  let op = OperatorFactory.memory()
+  let builder = new Memory()
+  let op = builder.build()
   let content = "hello world"
   let path = 'test'
 
-  await op.write(path, Array.from(new TextEncoder().encode(content)))
+  let o = op.object(path)
 
-  let meta = await op.meta(path)
-  t.is(meta.size, content.length)
+  await o.write(new TextEncoder().encode(content))
 
-  let res = await op.read(path)
-  t.is(content, new TextDecoder().decode(Buffer.from(res)))
+  let meta = await o.stat()
+  t.is(meta.mode, 0)
```

Comment:
Can we have enum?

Reply:
Javascript does not have an enum type. It can only use in a TypeScript file.

Reply:
Nowadays, TypeScript is very popular. At least, it can be useful.